app/src/sharedTest/java/org/oppia/app/recyclerview/BindableAdapterTest.kt

```diff
@@ -251,6 +251,62 @@ class BindableAdapterTest {
     }
   }
 
+  @Test
+  fun testBindableAdapter_incomingSameData_noRebindingShouldHappen() {
+
+    val adapter = createMultiViewTypeNoDataBindingBindableAdapter()
+    BindableAdapterTestFragmentPresenter.testBindableAdapter = adapter
+
+    lateinit var dataObserver: RecyclerView.AdapterDataObserver
+    var doesRebind = false
+
+    val oldList = listOf(STR_VALUE_1, STR_VALUE_1, INT_VALUE_1).toMutableList()
+    val newList = listOf(STR_VALUE_1, STR_VALUE_1, INT_VALUE_1).toMutableList()
+
+    ActivityScenario.launch(BindableAdapterTestActivity::class.java).use { scenario ->
+
+      scenario.onActivity { activity ->
+        dataObserver = object : RecyclerView.AdapterDataObserver() {
```

Comment:
Consider making this either a fake class or a mock to make it more readily reusable, and easier to verify as correct.

Reply:
Using mock now